template/lambda-sqs-worker-cdk/src/framework/handler.ts

```diff
@@ -38,3 +44,30 @@ export const createHandler = <Event, Output = unknown>(
       }
     }),
   );
+
+export const createBatchSQSHandler =
+  (
+    fn: (record: SQSRecord, ctx: LambdaContext) => Promise<unknown>,
+  ): Handler<SQSEvent, SQSBatchResponse> =>
+  async (event, ctx) => {
+    const processRecord = (
+      record: SQSRecord,
+    ): Promise<SQSBatchItemFailure | undefined> =>
+      recordContext.run({ sqsMessageId: record.messageId }, async () => {
+        try {
+          await fn(record, ctx);
```

Comment:
I won't block this PR as it's an improvement over the current state, but this model doesn't promote optimal I/O usage as you're still processing the records individually within the process, unless you try to have something clever like dataloader behind the scenes. Not sure if that's worth mentioning in the template, or if we'd prefer simplicity to be the default here 🤷